spec/re_captcha/configuration_spec.rb

```diff
@@ -26,20 +26,86 @@
     end
 
     describe '#valid?' do
+      let(:errors) { double('Array', :<< => nil) }
+
+      it 'resets errors' do
+        expect(instance).to receive(:reset_errors)
+        instance.valid?
+      end
+
+      it 'sets errors' do
+        expect(instance).to receive(:set_errors)
+        instance.valid?
+      end
+
+      it 'checks the number of errors' do
+        allow(instance).to receive(:errors).and_return(errors)
+        expect(errors).to receive(:empty?)
+        instance.valid?
```

Comment:
this spec can be removed. It actually tests that the `valid?` returns `true` or `false` and you already got that covered in the next spec.

On a related note I'm a real asshole about expectations that mirror the method body :) . My goal with those changes are that if you for example decide to change the method to

```ruby
include Active::Model

def valid
  errors.add(:private_key, "missing") if private_key.nil?
  errors.add(:public_key, "missing") if public_key.nil?
  errors.any?
end
```

The specs will continue to pass because the behavior is exactly the same as before, and specs should test that - not the implementation details.